src/vm/tests/test-docker.js

```diff
@@ -1121,3 +1123,194 @@ test('test docker VM with good zlog_max_size', function (t) {
+                                time: ev.data.when
+                            });
+                        }
+                    }
+                }
+
+                if (starts >= num_cycles && stops >= num_cycles) {
+                    // stop the zoneevent watcher
+                    se.stop();
+                    se = null;
+                    emitter.emit('done');
+                }
+            });
+            cb();
+        }, function (cb) {
+            // start the VM
+            VM.start(state.uuid, {}, function (err) {
+                t.ok(!err, 'starting VM: ' + (err ? err.message : 'success'));
+                cb(err);
+            });
+        }, function (cb) {
+            // wait for our num_cycles to finish
+            emitter.once('done', function () {
+                var deltas = [];
+                var expected_counts = [];
+                var expected_delays = [];
+                var i;
+                var last_stop = 0;
+                var restartcounts = [];
+                var restartdelays = [];
+
+                events.forEach(function (evt) {
+                    if (evt.action === 'start') {
+                        if (last_stop > 0) {
+                            deltas.push((evt.time - last_stop) / 1000000);
+                        }
+                    } else if (evt.action === 'stop') {
+                        last_stop = evt.time;
+                    } else {
+                        throw (new Error('Unexpected action: ' + evt.action));
+                    }
+                });
+
+                restartcounts = events.filter(function (evt) {
+                    if (evt.action === 'stop') {
+                        return (true);
+                    }
+                    return (false);
+                }).map(function (evt) {
+                    return (evt.restartcount);
+                });
+
+                restartdelays = events.filter(function (evt) {
+                    if (evt.action === 'stop') {
+                        return (true);
+                    }
+                    return (false);
+                }).map(function (evt) {
+                    return (evt.restartdelay);
+                });
+
+                t.equal(trend(deltas.slice(0, cycles_fail - 1)), 1, 'first '
+                    + (cycles_fail - 1) + ' should go up');
+                t.equal(trend(deltas.slice(cycles_fail, deltas.length)), 1,
+                    ' last ' + (deltas.length - cycles_fail) + ' should go up');
+                t.ok((deltas[cycles_fail - 1] > deltas[cycles_fail]),
+                    deltas[cycles_fail - 1] + ' > ' + deltas[cycles_fail]);
+
+                // should be [0, 1, 2, ... 14]
+                for (i = 0; i < num_cycles; i++) {
+                    expected_counts.push(i);
+                }
+
+                // delays should be: [null, 200, 400 ... 12800, 200, ... 12800]
+                for (i = 0; i < cycles_fail; i++) {
```

Comment:
Same as above: adding some of the content from the PR's description that describes the logic of the test would help understand why these values are expected.